src/test/java/BaseTest.java

```diff
@@ -1,18 +1,108 @@
+import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
-import org.openqa.selenium.firefox.FirefoxDriver;
-import org.openqa.selenium.safari.SafariDriver;
-import org.testng.annotations.BeforeMethod;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.annotations.*;
+import java.time.Duration;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.openqa.selenium.interactions.Actions;
 import org.testng.annotations.BeforeSuite;
-
+import static org.openqa.selenium.By.cssSelector;
+import static org.openqa.selenium.By.xpath;
 
 public class BaseTest {
-
+    public static WebDriver driver = null;
+    public static WebDriverWait wait;
+    public static String url = null;
+    public static Actions actions;
+    String newPlayListName = "New Playlist";
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    @BeforeMethod
+    @Parameters ({"BaseURL"})
+    public static void openBrowser(String BaseURL) {
+        ChromeOptions option = new ChromeOptions();
+        option.addArguments("--remote-allow-origins=*");
+        driver = new ChromeDriver(option);
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        url= BaseURL;
+        driver.get(url);
+    }
+
+    public void logIn() {
+        enterEmail("linulya1411@gmail.com");
+        enterPassword("te$t$tudent");
+        clickSubmit();
+    }
+    By enterEmailAddress = cssSelector(("[type = 'email']"));
+    By passwordField = cssSelector(("[type='password']"));
+    By submitButton = cssSelector(("[type='submit']"));
+    By playList = By.xpath(("//*[@id=\"playlists\"]/ul/li[3]/a"));
+    By xButton = cssSelector(("button[title='Delete this playlist']"));
+    By notificationMessage = cssSelector(("div.success.show"));
+    By editOption = cssSelector(("[data-testid='playlist-context-menu-edit-48111']"));
+    By textField = cssSelector(("input[name='name']"));
+    By newPlayList = By.xpath(("//a[text()='"+newPlayListName+"']"));
+
+    public void enterEmail(String email) {
+        WebElement emailElement = wait.until(ExpectedConditions.elementToBeClickable(enterEmailAddress));
+        emailElement.sendKeys("linulya1411@gmail.com");
+    }
+    public void enterPassword(String password) {
+        WebElement passwordElement = wait.until(ExpectedConditions.elementToBeClickable(passwordField));
+        passwordElement.sendKeys("te$t$tudent");
+    }
+
+    public void clickSubmit() {
+        WebElement submitElement = wait.until(ExpectedConditions.elementToBeClickable(submitButton));
+        submitElement.click();
+    }
+    public void clickPlayList() {
+        WebElement clickOnPlaylist = wait.until(ExpectedConditions.visibilityOfElementLocated(playList));
+        actions.contextClick(clickOnPlaylist).perform();
```

Comment:
You got this problem, please correct it. 

![image](https://user-images.githubusercontent.com/104761482/224565516-428ad9b4-f087-448a-a4c2-b5deb5179016.png)

**Hint**

1.

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >

<suite name="TestNG">
    <parameter name = "BaseURL" value = "https://bbb.testpro.io/"/>
    <test name="Example Test" preserve-order="false">
        <classes>
            <class name="Homework21">
            </class>
        </classes>
    </test>
</suite>
```

2.

```
    @BeforeMethod
    @Parameters ({"BaseURL"})
    public static void openBrowser(@Optional String BaseURL) {
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--disable-notifications","--remote-allow-origins=*", "--incognito","--start-maximized");
        driver = new ChromeDriver(options);
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
        actions = new Actions(driver);
        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
        url= BaseURL;
        driver.get(url);
    }
```